aries_cloudagent/connections/models/diddoc_v2/diddoc.py

```diff
@@ -239,11 +268,99 @@ def id(self, value: str) -> None:
 
         self._id = value
 
-    def set(
-        self,
-        item: Union[Service, VerificationMethod],
-        upsert=False,
-        verification_type="publicKey",
+    def add_verification_method(
+            self,
+            type: Union[PublicKeyType, str],
+            controller: Union[str, Sequence],
+            value: str,
+            *,
+            ident: str = None,
+            usage: str = None,
+            authentication: bool = False,
+            verification_type: str = "publicKey",
+            upsert: bool = False
+    ) -> VerificationMethod:
+        """Add a verification method to this document."""
+        if ident:
+            id = "{}#{}".format(self.id, ident)
+            if self._index.get(id) and not upsert:
+                raise ValueError("ID already exists, use arg upsert to update it")
+        else:
+            for index in range(1, 100):
+                id_aux = "{}#keys-{}".format(self.id, index)
+                if not self._index.get(id_aux):
+                    id = id_aux
+                    break
+
+        key = VerificationMethod(id, type, controller, usage, value, authentication)
+        self._set(key, upsert=upsert, verification_type=verification_type)
+        return key
+
+    def add_service(
+            self,
+            type: Union[str, list],
+            endpoint: Union[str, Sequence, dict],
+            ident: str = None,
+            priority: int = 0,
```

Comment:
`priority` is unique to the didcomm service type so it doesn't make sense to include it in this method. Instead, `kwargs` with arbitrary additional parameters should be allowed and added to the service. This might require slightly loosening our Service schema and adding a way for accessing these arbitrary additional parameters to the object representing services. Due to the differences between generic services and DIDComm services, it likely makes sense to make a specialized `DIDCommService` object with a more restrictive schema and `recipient_keys`, `routing_keys`, and `priority` beyond the base required `type`, `id`, and `endpoint`.